src/main/java/org/apache/mesos/scheduler/recovery/SimpleRecoveryPlanManager.java

```diff
@@ -0,0 +1,325 @@
+package org.apache.mesos.scheduler.recovery;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.TextFormat;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.mesos.Protos;
+import org.apache.mesos.offer.DefaultOfferRequirementProvider;
+import org.apache.mesos.offer.InvalidRequirementException;
+import org.apache.mesos.offer.OfferRequirementProvider;
+import org.apache.mesos.scheduler.plan.*;
+import org.apache.mesos.scheduler.recovery.constrain.LaunchConstrainer;
+import org.apache.mesos.scheduler.recovery.monitor.FailureMonitor;
+import org.apache.mesos.specification.DefaultTaskSpecification;
+import org.apache.mesos.specification.InvalidTaskSpecificationException;
+import org.apache.mesos.state.StateStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * {@link SimpleRecoveryPlanManager} enables monitoring and management of recovery plan.
+ *
+ * This is an implementation of {@code PlanManager} that performs task recovery using dynamically generated
+ * {@code Plan}. {@link SimpleRecoveryPlanManager} tracks currently failed (permanent) and stopped (transient) tasks,
+ * generates a new {@link DefaultRecoveryBlock} for them and adds them to the recovery Plan, if not already added.
+ */
+public class SimpleRecoveryPlanManager implements PlanManager {
```

Comment:
So far the default/sample implementations have been named `DefaultFoo`, would it make sense to continue that pattern here?